test/jdk/java/awt/PrintJob/ImageTest/ImageTest.java

```diff
@@ -0,0 +1,164 @@
+                Dimension dim = pjob.getPageDimension();
+                if (width > dim.width) {
+                    width = dim.width - 30; // take care of paper margin
+                }
+                if (height > dim.height) {
+                    height = dim.height - 30;
+                }
+            }
+            g.drawImage(img, 10, 75, width, height, this);
+        }
+
+        public void setPrintJob(PrintJob pj) {
+            pjob = pj;
+        }
+
+        public boolean imageUpdate(Image img, int infoflags,
+                                   int x, int y, int w, int h) {
+            if ((infoflags & ALLBITS) != 0) {
+                repaint();
+                return false;
+            }
+            return true;
+        }
+    }
+
+    private static Frame init() {
+        ImageFrame f = new ImageFrame();
+        f.setLayout(new FlowLayout());
+        Button b = new Button("Print");
+        b.addActionListener(e -> {
+            PrintJob pj = Toolkit.getDefaultToolkit()
+                    .getPrintJob(f, "ImageTest", null);
+            if (pj != null) {
+                f.setPrintJob(pj);
+                Graphics pg = pj.getGraphics();
+                f.paint(pg);
+                pg.dispose();
+                pj.end();
+            }
+        });
+        f.add(b);
+        f.setBounds(0, 50, 700, 350);
+
+        return f;
+    }
+
+    private static void createImage() throws IOException {
+        final BufferedImage bufferedImage =
+                new BufferedImage(600, 230, BufferedImage.TYPE_INT_RGB);
+        Graphics2D g2d = bufferedImage.createGraphics();
+
+        g2d.setColor(new Color(0xE7E7E7));
+        g2d.fillRect(0, 0, bufferedImage.getWidth(), bufferedImage.getHeight());
+
+        g2d.setColor(Color.YELLOW);
+        g2d.fillRect(0, 6, 336, 40);
+        g2d.setColor(Color.BLACK);
+        g2d.drawString("Yellow rectangle", 10, 30);
+
+        g2d.setColor(Color.CYAN);
+        g2d.fillRect(132, 85, 141, 138);
+        g2d.setColor(Color.BLACK);
+        g2d.drawString("Cyan rectangle", 142, 148);
+
+        g2d.setColor(Color.MAGENTA);
+        g2d.fillRect(432, 85, 141, 138);
+        g2d.setColor(Color.BLACK);
+        g2d.drawString("Magenta rectangle", 442, 148);
+
+        g2d.dispose();
+
+        ImageIO.write(bufferedImage, "gif", new File("image.gif"));
+    }
+
+    private static final String INSTRUCTIONS = """
+             Click the Print button on the Frame. Select a printer from the
+             print dialog and click 'OK'. Verify that the image displayed
+             in the Frame is correctly printed. Test printing in both Color
+             and Monochrome.
+            """;
+
+    public static void main(String[] args) throws Exception {
+        if (PrinterJob.lookupPrintServices().length == 0) {
+            throw new SkippedException("Printer not configured or available."
+                    + " Test cannot continue.");
```

Comment:
```suggestion
            throw new SkippedException("Printer not configured or available.");
```
We decided to shorten the message. Let's keep it consistent.